core/runtime/src/main/java/io/quarkiverse/langchain4j/guardrails/GuardrailsUtils.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkiverse.langchain4j.guardrails;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+class GuardrailsUtils {
+    private GuardrailsUtils() {
+    }
+
+    private static class MapperHolder {
+        private static final ObjectMapper MAPPER = new ObjectMapper();
+    }
+
+    static String trimNonJson(String llmResponse) {
+        int jsonStart = llmResponse.indexOf("{");
+        int jsonEnd = llmResponse.lastIndexOf("}");
+        if (jsonStart >= 0 && jsonEnd >= 0 && jsonStart < jsonEnd) {
+            return llmResponse.substring(jsonStart, jsonEnd + 1);
+        }
+        return null;
+    }
+
+    static boolean validateJson(String json, Class<?> expectedOutputClass) {
+        try {
+            MapperHolder.MAPPER.readValue(json, expectedOutputClass);
```

Comment:
You are immediately going into Databind. There are various issues:

1. I believe you need to verify it's valid JSON before mapping (see the example from the doc)
2. You pass a Class instead of a Type/TypeReference, so you won't handle generics and collections 
3. Mapping can be super expensive, and Quarkus is going to do it anyway latter anyway

Reply:
> You are immediately going into Databind. There are various issues:
> 
>     1. I believe you need to verify it's valid JSON before mapping (see the example from the doc)

Mapping is a stronger constraint and then a super set of a valid JSON. I was thinking that in this way I'm doing both things in one shoot, but I'm no longer sure if this is a good idea. However if the JSON is valid (and I guess that in most cases it will be) I will have to also perform a mapping anyway.

>     2. You pass a Class instead of a Type/TypeReference, so you won't handle generics and collections

Good point, I will fix this.
 
>     3. Mapping can be super expensive, and Quarkus is going to do it anyway latter anyway

That's another good point, but with more complex solution. Ideally if the mapping is successful we may want to pass downstream the deserialized object instead of having to redo the mapping once again, but at the moment output guardrails don't allow to do this. I will check if I can change them a bit to make this possible.

Reply:
Totally agree with your last point - if the mapping is successful we should pass that object as response.

Reply:
> yes that would be great specially in terms of performance

I already implemented this with my latest commit, give a look.